frontend/src/metabase/parameters/utils/dashboards.ts

```diff
@@ -139,12 +138,15 @@ function buildFieldFilterUiParameter(
   );
   const mappedFields = mappingsForParameter.map(mapping => {
     const { target, card } = mapping;
-    const question = new Question(card, metadata);
 
     try {
       const field = getTargetFieldFromCard(target, card, metadata);
 
-      return { field, shouldResolveFkField: !question.isNative() };
+      return {
+        field,
+        // The `dataset_query` is null for questions on a dashboard the user doesn't have access to
+        shouldResolveFkField: Boolean(card.dataset_query?.type !== "native"),
```

Comment:
no need for `Boolean`

Reply:
`shouldResolveFkField` expects a boolean value. If you pass `undefined` of `null` TS complains.

Reply:
Uladzimir is right :+1: 

----

`shouldResolveFkField` should probably evaluate to `false` when `dataset_query` is `null`.
We can also inverse the check for shorter notation.

```suggestion
        shouldResolveFkField: card.dataset_query?.type === "query",
```